### ISSUE TITLE
Consider adding source linking

### ISSUE THREAD

**Issue by paulomorgado** [rabbitmq/rabbitmq-dotnet-client#480]
With source linking it will be possible to debug in edge case scenarios.

I have a scenario where I'm ending up [here](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/master/projects/client/RabbitMQ.Client/src/client/impl/TcpClientAdapter.cs#L33) but calling this directly:

```
var adds = await Dns.GetHostAddressesAsync(host).ConfigureAwait(false);
var ep = TcpClientAdapterHelper.GetMatchingHost(adds, AddressFamily.InterNetwork);
```

yields an `IpAddress`.

**Comment by michaelklishin** [comment]
Can this be expanded a bit? I'm not sure what specifically is being asked here.

**Comment by bording** [comment]
It would mean adding https://github.com/dotnet/sourcelink#githubcom-and-github-enterprise to the project.

This would let you directly step into the source of the client when debugging.

**Comment by bording** [comment]
To fully take advantage of everything the package provides, there might need to be some changes to how the NuGet packages are created.

**Comment by paulomorgado** [comment]
What that package is doing to support source linking is creating a JSON file for the compiler to embed in the PDB: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/pathmap-compiler-option

Additionally, it supports the new NuGet metadata for the repository: https://blog.nuget.org/20180827/Introducing-Source-Code-Link-for-NuGet-packages.html

It can definitely be done by that package.

The Microsoft / .NET Foundation package is still in beta although it is already being widely used ([Newtowsoft.Json](https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Newtonsoft.Json.csproj#L37)). But the original ones by the original creator are still available.

**Comment by lukebakken** [comment]
This will be implemented in #697 , thanks.
